app/src/main/java/com/github/dedis/student20_pop/utility/network/WebSocketLowLevelProxy.java

```diff
@@ -240,29 +214,32 @@ public void purge() {
 
     private Optional<Session> getSession() {
         try {
-            return Optional.ofNullable(future2.getNow(null));
+            return Optional.ofNullable(sessionUse.getNow(null));
         } catch (Throwable t) {
             return Optional.empty();
         }
     }
 
-    /**
-     * Close the socket for the given reason.
-     *
-     * @param reason of the closing
-     */
+    @Override
     public void close(Throwable reason) {
         synchronized (SESSION_LOCK) {
-            future2.completeExceptionally(reason);
+            sessionUse.completeExceptionally(reason);
 
-            future1.thenAccept(s -> {
+            session.thenAccept(s -> {
                 try {
                     s.close();
                 } catch (IOException e) {
                     e.printStackTrace();
                 }
             });
         }
+
+        // Complete all pending requests and remove them
+        Iterator<Map.Entry<Integer, RequestEntry>> it = requests.entrySet().iterator();
```

Comment:
What does the iterator `it` represent? Maybe you can rename it to something more helpful

Reply:
it for iterator